GiniVision/Classes/OpenWithTutorialViewController.swift

```diff
@@ -0,0 +1,117 @@
+            UIImageNamedPreferred(named: "openWithTutorialStep3"))
+    ]
+    
+    lazy var headerTitle: String = {
+        let localizedString = NSLocalizedStringPreferred("ginivision.help.openWithTutorial.collectionHeader", comment: "intoduction header for further steps")
+        return String(format: localizedString, self.appName)
+    }()
+    
+    fileprivate var stepsCollectionLayout: OpenWithTutorialCollectionFlowLayout {
+        return self.collectionView?.collectionViewLayout as! OpenWithTutorialCollectionFlowLayout
+    }
+    
+    public init() {
+        super.init(collectionViewLayout: OpenWithTutorialCollectionFlowLayout())
+    }
+    
+    required public init?(coder aDecoder: NSCoder) {
+        fatalError("init() should be called instead")
+    }
+    
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        self.title = NSLocalizedString("ginivision.help.openWithTutorial.title", bundle: Bundle(for: GiniVision.self), comment: "title shown when the view controller is within a view controller")
+        self.view.backgroundColor = Colors.Gini.pearl
+        self.collectionView!.backgroundColor = nil
+        self.edgesForExtendedLayout = []
+        self.automaticallyAdjustsScrollViewInsets = false
+
+        self.collectionView!.register(OpenWithTutorialCollectionCell.self, forCellWithReuseIdentifier: openWithTutorialCollectionCellIdentifier)
+        self.collectionView!.register(OpenWithTutorialCollectionHeader.self, forSupplementaryViewOfKind: UICollectionElementKindSectionHeader, withReuseIdentifier: openWithTutorialCollectionHeaderIdentifier)
+        
+        stepsCollectionLayout.minimumLineSpacing = 1
+        stepsCollectionLayout.minimumInteritemSpacing = 1
+        stepsCollectionLayout.estimatedItemSize = estimatedCellSize(widthParentSize: view.frame.size)
+    }
+    
+    public override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        
+        coordinator.animate(alongsideTransition: { [unowned self] _ in
+            self.stepsCollectionLayout.estimatedItemSize = self.estimatedCellSize(widthParentSize: size)
+            self.collectionView?.collectionViewLayout.invalidateLayout()
+        })
+    }
+    
+    private func estimatedCellSize(widthParentSize size: CGSize) -> CGSize {
+        if size.width > size.height && UIDevice.current.isIpad {
+            let width:CGFloat = round(UIScreen.main.bounds.width / CGFloat(self.items.count) - CGFloat(self.stepsCollectionLayout.minimumInteritemSpacing * CGFloat(self.items.count - 1)))
+            return CGSize(width: width, height: UIScreen.main.bounds.height)
+        } else {
+            return CGSize(width: UIScreen.main.bounds.width, height: 550)
```

Comment:
Why not use the view controller view's frame? You'd lose an external dependency.
Also, it would be more correct. When I see something like this, I suspect that there was a problem with changing the view controller size.

Reply:
(https://media.giphy.com/media/U1XhGr8CWqvVC/giphy.gif). I probably moved that from other method and I forgot to change it. 👍